canonical-web-and-design/ubuntu.com
# Add preview feature for inactive engage pages

These are the rules:
1. If you go on `/engage`. By default you will see no inactive pages. To see the inactive pages you have to go to `/engage?preview`. This will set the meta `robot` to `noindex`.
2. If you go to an inactive engage page you will be redirected to its discourse page. If you can visit the page if you add the flag `?preview`. This will set the meta `robot` of the page to `nofollow`.
3. If you are `/engage?preview` all inactive pages will have their URL changed to `<URL>?preview`

## QA

For the sake of the QA I made `/engage/it/deployment-azienda-manuale` `active = false`. It will not affect prod in any way because the feature is not live yet.

1. Go to https://ubuntu-com-8761.demos.haus/engage
You will not see **Cinque strategie per accelerare l'implementazione di Kubernetes nell'impresa** listed. It should be first if you check https://ubuntu.com/engage.
2. Press `ctrl + u` in browser. Use search to look for the meta tag `robot`. It won't be there.
3. Add the `?preview` flag (https://ubuntu-com-8761.demos.haus/engage?preview). You will see the post **Cinque strategie per accelerare l'implementazione di Kubernetes nell'impresa** show up first.
4. Press `ctrl + u` in browser. Use search to look for the meta tag `robot`. You will see it has the value `noindex`.
6. Hover over the title URL of the post you will see the `?preview` added at the end of the link. Also check that active pages do not have it.
7. Go to https://ubuntu-com-8761.demos.haus/engage/it/deployment-azienda-manuale you will be redirected to its discourse page.
8. Go to https://ubuntu-com-8761.demos.haus/engage/it/deployment-azienda-manuale?preview you will see the page
9. Press `ctrl + u` in browser. Use search to look for the meta tag `robot`. You will see it has the value `nofollow`.
10. Go to https://ubuntu-com-8761.demos.haus/engage/it/redhat-openstack-confronto-manuale (an active page). Press `ctrl + u` in browser. Use search to look for the meta tag `robot`. It won't be there.

## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/3404
